ls1intum/Artemis
# `Adaptive learning`: Improve learning path management

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


This PR addresses improvements related to the management of Learning Paths in Artemis.
Stacked on #6747 .


### Description

- fixes several style issues
- fixes position of Learning Paths tab in `course management-tab-bar`
- adds endpoint to retrieve health status of learning paths in course
- adds endpoint to generate missing learning paths in course
- adds card in learning path management to display health status warning
- adds more information and navigation options in the details view of learning paths

### Steps for Testing

Prerequisites:
- 1 Admin
- Course (enrollment allowed)
- 1 Instructor
- \>2 Students (the more the merrier, 1 enrolled, 1 not enrolled yet)
- 1 Competency with 1 Exercise and 1 Lecture Unit

1. Log in to Artemis as Instructor
2. Navigate to Course Administration -> Learning Paths
3. You should see a message that learning paths are not enabled yet
4. Click the `Enable Learning Paths` button
5. The content should load the student's learning paths and display them as a pageable list
6. Make sure there is a Learning Path for each enrolled student
7. Try to search for students
8. Click on the button to view a student's Learning Path
9. A modal should open that displays the Learning Path Graph
10. Drag and Zoom the Graph and test if the Navigation-Buttons on top work as expected
11. Log in to Artemis as a unenrolled Student
12. Enroll in the course
13. Log in to Artemis as Instructor
14. Make sure the newly enrolled student's Learning Path is contained in the List
15. Manually add a user to the student group of your course
16. Make sure the student's learning path is now on the list under Learning Path Management

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage







### Screenshots


Learning Path Management:
<img width="1721" alt="Screenshot 2023-07-29 at 15 04 04" src="https://github.com/ls1intum/Artemis/assets/44003963/1c8d0973-67e7-488f-b06f-5a531640324f">

Progress/Details view:
<img width="1163" alt="Screenshot 2023-07-29 at 15 45 34" src="https://github.com/ls1intum/Artemis/assets/44003963/2787470c-5364-4a83-b1c1-6dee90506772">
